adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/FlagStat.scala

```diff
@@ -93,15 +94,15 @@ object FlagStat {
         val (primaryDuplicates, secondaryDuplicates) = DuplicateMetrics(p)
         new FlagStatMetrics(1,
           primaryDuplicates, secondaryDuplicates,
-          b2i(p.getReadMapped),
-          b2i(p.getReadPaired),
-          b2i(p.getReadPaired && p.getFirstOfPair),
-          b2i(p.getReadPaired && p.getSecondOfPair),
-          b2i(p.getReadPaired && p.getProperPair),
-          b2i(p.getReadPaired && p.getReadMapped && p.getMateMapped),
-          b2i(p.getReadPaired && p.getReadMapped && !p.getMateMapped),
-          b2i(mateMappedToDiffChromosome),
-          b2i(mateMappedToDiffChromosome && p.getMapq >= 5),
+          b2i(b(p.getReadMapped)),
```

Comment:
Yea, the genesis of this was that I was running `flagstat` on my paired-fastq-converted `.adam`s and it was NPEing on several of these fields which were [left unset or set to `null` by `FastqRecordConverter.convert`](https://github.com/bigdatagenomics/adam/pull/831/files#diff-eb57d01d374e75a94641ec53a4b62fa5R117).

So given that my paired-fastq-`.adam`s were unable pass a basic read-counting verification step post-conversion, I thought this functionality should be included in this PR.

lmk if you still think it should be in a separate PR, I can go either way.